### PR TITLE
pcm: Replace ioctl() with pcm->ops->ioctl()

### PR DESCRIPTION
**Pull request by quic-qianch**
Add plugin support for pcm_ioctl according to the define TINYALSA_USES_PLUGINS

### CODE REVIEW

File: src/pcm.c [hunk 1]
```diff
@@ -1806,6 +1806,9 @@ int pcm_ioctl(struct pcm *pcm, int request, ...)
     arg = va_arg(ap, void *);
     va_end(ap);
 
-    // FIXME Does not handle plugins
+#ifdef TINYALSA_USES_PLUGINS
+    return pcm->ops->ioctl(pcm->data, request, arg);
```
Comment (quic-rohit) [inline src/pcm.c:1810]: pcm->ops->ioctl() is valid for pcm_hw.c as well. There is no need to have #ifdef check.

Comment (quic-rohit) [reply src/pcm.c:1810]: Can we also add support of pcm->ops->ioctl() at other places in the file.
e.g.
    if (ioctl(pcm->fd, SNDRV_PCM_IOCTL_DELAY, &pcm->pcm_delay) < 0)
        return -1;

File: src/pcm.c [hunk 2]
```diff
@@ -1806,6 +1806,9 @@ int pcm_ioctl(struct pcm *pcm, int request, ...)
     arg = va_arg(ap, void *);
     va_end(ap);
 
-    // FIXME Does not handle plugins
```
Comment (quic-rohit) [inline src/pcm.c:1809]: remove this FIXME

File: src/pcm.c [hunk 3]
```diff
@@ -1806,6 +1806,9 @@ int pcm_ioctl(struct pcm *pcm, int request, ...)
...
+#ifdef TINYALSA_USES_PLUGINS
```
Comment (quic-rohit) [inline src/pcm.c]: This change is not needed as it is handled in https://github.com/tinyalsa/tinyalsa/commit/03f333e3b41f1a0e582689eed440efc6b0c1daea . Please remove this change from list.

**Review by quic-rohit — APPROVED** [review]

**Review by dvdli — APPROVED** [review]

**Pull request merged**